### ISSUE TITLE
arcface_torch: There are periodic, sudden drops in losses

### ISSUE THREAD

**Issue by SuyueLiu** [deepinsight/insightface#2266]
![image](https://user-images.githubusercontent.com/49122192/226346190-5858b32a-fb08-4b11-8a01-5e3eb5922ac9.png)

backbone: mbf
loss: arcface
dataset: glint (msra + asian celeb)
lr = 0.1 wd = 1e-4  fp16=True

as shown in the figure, there are periodic, sudden drops in losses, I am curious why this happens.

**Comment by Sunshine066** [comment]
Ha ha, I have the same question as you

**Comment by SuyueLiu** [comment]
> Ha ha, I have the same question as you. But my cyclical decline is followed by a small sharp rise followed by a steady decline

Same as you, did you solve this problem?

**Comment by 411104983** [comment]
The same is true for the loss when I train the VIT model, but the final result does not seem to be worse

**Comment by Sunshine066** [comment]
> The same is true for the loss when I train the VIT model, but the final result does not seem to be worse

Yes, it doesn't make any difference to the final result, but I wonder how it works? I don't have this problem when I train other networks

**Comment by Sunshine066** [comment]
The loss changes periodically with the epoch. At the beginning of each epoch, the loss drops precipitously and increases slowly in the later training process. I do not understand the reason for this phenomenon.Haha,is there expert know the reason?

**Comment by HHCorp** [comment]
> The loss changes periodically with the epoch. At the beginning of each epoch, the loss drops precipitously and increases slowly in the later training process. I do not understand the reason for this phenomenon.Haha,is there expert know the reason?

#377 This may answer your question.
